src/VisualStudio/Core/Def/EditorConfigSettings/Analyzers/View/SeverityControl.xaml.cs

```diff
@@ -0,0 +1,119 @@
+using System.Windows.Controls;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Data;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.Analyzers.View
+{
+    /// <summary>
+    /// Interaction logic for SeverityControl.xaml
+    /// </summary>
+    internal partial class SeverityControl : UserControl
+    {
+        private readonly ComboBox _comboBox;
+        private readonly AnalyzerSetting _setting;
+
+        public SeverityControl(AnalyzerSetting setting)
+        {
+            InitializeComponent();
+            var refactoring = CreateGridElement(KnownMonikers.None, ServicesVSResources.Disabled);
+            var suggestion = CreateGridElement(KnownMonikers.StatusInformation, ServicesVSResources.Suggestion);
+            var warning = CreateGridElement(KnownMonikers.StatusWarning, ServicesVSResources.Warning);
+            var error = CreateGridElement(KnownMonikers.StatusError, ServicesVSResources.Error);
+            _comboBox = new ComboBox()
+            {
+                ItemsSource = new[]
+                {
+                    refactoring,
+                    suggestion,
+                    warning,
+                    error
+                }
+            };
+
+            switch (setting.Severity)
+            {
+                case DiagnosticSeverity.Hidden:
+                    _comboBox.SelectedIndex = 0;
+                    break;
+                case DiagnosticSeverity.Info:
+                    _comboBox.SelectedIndex = 1;
+                    break;
+                case DiagnosticSeverity.Warning:
+                    _comboBox.SelectedIndex = 2;
+                    break;
+                case DiagnosticSeverity.Error:
+                    _comboBox.SelectedIndex = 3;
+                    break;
+                default:
+                    break;
+            }
+
+            _comboBox.SelectionChanged += ComboBox_SelectionChanged;
+
+            _ = RootGrid.Children.Add(_comboBox);
+            _setting = setting;
+        }
+
+        private void ComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            switch (_comboBox.SelectedIndex)
+            {
+                case 0:
+                    _setting.ChangeSeverity(DiagnosticSeverity.Hidden);
+                    return;
+                case 1:
+                    _setting.ChangeSeverity(DiagnosticSeverity.Info);
+                    return;
+                case 2:
+                    _setting.ChangeSeverity(DiagnosticSeverity.Warning);
+                    return;
+                case 3:
+                    _setting.ChangeSeverity(DiagnosticSeverity.Error);
+                    return;
+                default: return;
+            }
+        }
+
+        private static FrameworkElement CreateGridElement(ImageMoniker imageMoniker, string text)
```

Comment:
This function name makes less sense in this context. 
```suggestion
        private static FrameworkElement CreateItemElement(ImageMoniker imageMoniker, string text)
```